src/messages.cpp

```diff
@@ -1037,15 +1070,23 @@ namespace quicr::messages {
     Bytes& operator<<(Bytes& buffer, const MoqStreamSubGroupObject& msg)
     {
         buffer << UintVar(msg.object_id);
-        PushExtensions(buffer, msg.extensions);
-        buffer << UintVar(msg.payload.size());
-        buffer << msg.payload;
+        if (!msg.payload.size()) {
+            // empty payload needs a object status to be set
+            buffer << UintVar(msg.payload.size());
+            buffer << UintVar(static_cast<uint8_t>(msg.object_status));
+            PushExtensions(buffer, msg.extensions);
+        } else {
+            buffer << UintVar(msg.payload.size());
+            PushExtensions(buffer, msg.extensions);
+            buffer << msg.payload;
+        }
```

Comment:
Payload size will be written anyways
```suggestion
        buffer << UintVar(msg.payload.size());
        if (msg.payload.empty()) {
            // empty payload needs a object status to be set
            buffer << UintVar(static_cast<uint8_t>(msg.object_status));
            PushExtensions(buffer, msg.extensions);
        } else {
            PushExtensions(buffer, msg.extensions);
            buffer << msg.payload;
        }
```